src/web/src/app/space/service.ts

```diff
@@ -53,4 +141,158 @@ export class SpaceService {
+		const res: PaginatorApi<Space> = {
+			count: this.spaces.length,
+			results: this.spaces.slice(limit * page, limit * page + limit),
+		};
+		return of(res);
+	}
+
+	getSpaceListAll() {
+		const res: PaginatorApi<Space> = {
+			count: this.spaces.length,
+			results: this.spaces,
+		};
+		return of(res);
+	}
+
+	getSpaceNameById(spaceId: string) {
+		let space = this.spaces.find(s => s.id === spaceId);
+		return of(space?.name);
+	}
+
+	getSpaceById(spaceId: string) {
+		let space = this.spaces.find(s => s.id === spaceId);
+		return of(space);
+	}
+
+	getInvitesList(limit: number, page: number): Observable<PaginatorApi<SpaceInvitation>> {
+		const res: PaginatorApi<SpaceInvitation> = {
+			count: this.invites.length,
+			results: this.invites.slice(limit * page, limit * page + limit),
+		};
+		return of(res);
+	}
+
+	getInvitesListForSpace(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceInvitation>> {
+
+		let grouped: SpaceInvitation[] = [];
+		this.invites.forEach(function(invite) {
+			if (invite.spaceId === spaceId){
+				grouped.push(invite);
+			}
+		});
+		const res: PaginatorApi<SpaceInvitation> = {
+			count: grouped.length,
+			results: grouped.slice(limit * page, limit * page + limit),
+		};
+		return of(res);
+
+	}
+
+	getLinksListForSpace(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceLink>> {
+
+		let grouped: SpaceLink[] = [];
+		this.links.forEach(function(link) {
+			if (link.spaceId === spaceId){
+				grouped.push(link);
+			}
+		});
+		const res: PaginatorApi<SpaceLink> = {
+			count: grouped.length,
+			results: grouped.slice(limit * page, limit * page + limit),
+		};
+		return of(res);
+
+	}
+
+	getUsersListForSpace(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceUser>> {
+
+		let grouped: SpaceUser[] = [];
+		this.users.forEach(function(user) {
+			if (user.spaceId === spaceId){
+				grouped.push(user);
+			}
+		});
+		const res: PaginatorApi<SpaceUser> = {
+			count: grouped.length,
+			results: grouped.slice(limit * page, limit * page + limit),
+		};
+		return of(res);
+
+	}
+
+	getLinksList(limit: number, page: number): Observable<PaginatorApi<SpaceLink>> {
+		const res: PaginatorApi<SpaceLink> = {
+			count: this.links.length,
+			results: this.links.slice(limit * page, limit * page + limit),
+		};
+		return of(res);
+	}
+
+	isSpaceExists(key: string) {
+		return this.spaces.some(s => s.key === key);
+	}
+
+	createNewSpace(name: string, key: string, requestsAllowed: boolean) {
+		this.spaces.push({
+			id: crypto.randomUUID(),
+			name: name,
+			key: key,
+			requestsAllowed: requestsAllowed,
+			createdAt: new Date()
+		});
+	}
+
+	createNewInvite(spaceId: string, userId: string, role: string, creatorId: string) {
+		this.invites.push({
+			id: Math.random(),
+			spaceId: spaceId,
+			userId: userId,
+			role: role,
+			creatorId: creatorId,
+			createdAt: new Date(),
+		});
+	}
```

Comment:
Эти методы асинхронные, по тому как дёргают бэк для складывания записей в БД. Давай добавим небольшую задержку чтобы нагляднее этот момент обработать на фронте.
```suggestion
	createNewInvite(spaceId: string, userId: string, role: string, creatorId: string): Observable<boolean> {
		this.invites.push({
			id: Math.random(),
			spaceId: spaceId,
			userId: userId,
			role: role,
			creatorId: creatorId,
			createdAt: new Date(),
		});
		return of(true)
			.pipe(delay(2000));
	}
```

Reply:
Готово.